src/dc_tools.rs

```diff
@@ -1528,29 +1538,57 @@ impl<T: AsRef<std::ffi::OsStr>> OsStrExt for T {
 #[allow(dead_code)]
 fn os_str_to_c_string_unicode(
     os_str: &dyn AsRef<std::ffi::OsStr>,
-) -> Result<std::ffi::CString, CStringError> {
+) -> Result<CString, CStringError> {
     match os_str.as_ref().to_str() {
-        Some(val) => std::ffi::CString::new(val.as_bytes()).map_err(|err| match err {
+        Some(val) => CString::new(val.as_bytes()).map_err(|err| match err {
             std::ffi::NulError { .. } => CStringError::InteriorNullByte,
         }),
         None => Err(CStringError::NotUnicode),
     }
 }
 
-pub fn to_cstring<S: AsRef<str>>(s: S) -> std::ffi::CString {
-    std::ffi::CString::new(s.as_ref()).unwrap()
+/// Needs to free the result after use!
+pub unsafe fn to_cstring<S: AsRef<str>>(s: S) -> *mut libc::c_char {
+    let cstr = CString::new(s.as_ref()).expect("invalid string converted");
+    dc_strdup(cstr.as_ref().as_ptr())
```

Comment:
I'd much prefer if this got rebased on #204 where i replaced this badly named function with a StrExt trait with .strdup() method.  I think the solution there is more readable and less error-prone.  There's obviously multiple places where this comes up, but I'll leave it to this one comment.